lib/matplotlib/lines.py

```diff
@@ -788,10 +792,13 @@ def draw(self, renderer):
 
                 alt_marker_path = marker.get_alt_path()
                 if alt_marker_path:
-                    if rgbaFaceAlt is not None:
-                        gc.set_alpha(rgbaFaceAlt[3])
                     alt_marker_trans = marker.get_alt_transform()
                     alt_marker_trans = alt_marker_trans.scale(w)
+                    if (is_string_like(mec) and mec == 'auto' and
+                            rgbaFaceAlt is not None):
+                        gc.set_alpha(rgbaFaceAlt[3])
+                    else:
+                        gc.set_alpha(self.get_alpha())
```

Comment:
No.  It's just 4 lines.  To avoid passing in lots of things it would have to be a local function, defined inside the `draw` method.  You would end up with less readability.  If almost all of the marker "draw" functionality could be encapsulated in such a way that the same function could be used for the "regular" part and then called again for the "alt" part, that might be a good refactoring.  But even that is probably not worth fiddling with now, and certainly not in the scope of this PR.